src/call.ts

```diff
@@ -994,3 +996,43 @@ export const cancelOrders = async ({
     }, [] as CurrencyFlow[]),
   }
 }
+
+export const openVault = async ({
+  chainId,
+  userAddress,
+  inputToken,
+  outputToken,
+  options,
+}: {
+  chainId: CHAIN_IDS
+  userAddress: `0x${string}`
+  inputToken: `0x${string}`
+  outputToken: `0x${string}`
+  options?: DefaultOptions
+}): Promise<Transaction | undefined> => {
+  const publicClient = createPublicClient({
+    chain: CHAIN_MAP[chainId],
+    transport: options?.rpcUrl ? http(options.rpcUrl) : http(),
+  })
+  const pool = await fetchPool(
+    publicClient,
+    chainId,
+    [inputToken, outputToken],
+    !!(options && options.useSubgraph),
+  )
+  if (!pool.isOpened) {
+    return buildTransaction(
+      publicClient,
+      {
+        chain: CHAIN_MAP[chainId],
+        address: CONTRACT_ADDRESSES[chainId]!.Rebalancer,
+        account: userAddress,
+        abi: REBALANCER_ABI,
+        functionName: 'open',
+        args: [pool.market.bidBook.id, pool.market.askBook.id],
```

Comment:
p3; Is it working? As I know, 'open' function in Rebalancer has following interface
```
function open(IBookManager.BookKey calldata bookKeyA, IBookManager.BookKey calldata bookKeyB, address strategy)
```